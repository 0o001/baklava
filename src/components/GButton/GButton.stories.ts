import { storiesOf } from '@storybook/vue';
import GButton from './GButton.vue';

storiesOf('GButton', module)
  .addParameters({ component: GButton })
  .add('Base', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton size="big">Base Button</GButton>
        <GButton size="medium">Base Button</GButton>
        <GButton size="small">Base Button</GButton>
        <GButton outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('secondary', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="secondary">Base Button</GButton>
        <GButton variant="secondary" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('tertiary', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="tertiary">Base Button</GButton>
        <GButton variant="tertiary" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('light', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="light">Base Button</GButton>
      </div>
    `,
  }))
  .add('passive', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
        <GButton variant="passive">Base Button</GButton>
      </div>
    `,
  }))
  .add('success', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="success">Base Button</GButton>
      <GButton variant="success" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('warning', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="warning">Base Button</GButton>
      <GButton variant="warning" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('danger', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="danger">Base Button</GButton>
      <GButton variant="danger" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('info', () => ({
    components: { GButton },
    template: `
      <div class="g-p-10">
      <GButton variant="info">Base Button</GButton>
      <GButton variant="info" outline>Base Button</GButton>
      </div>
    `,
  }))
  .add('rightIcon', () => ({
    components: { GButton },
    template: '<GButton variant="primary" rightIcon="info">Base Button</GButton>',
  }))
  .add('leftIcon', () => ({
    components: { GButton },
    template: `
      <div>
        <GButton size="big" leftIcon="info">Left Icon</GButton>
        <GButton size="medium" leftIcon="info">Left Icon</GButton>
        <GButton size="small" leftIcon="info">Left Icon</GButton>
      </div>
    `,
  }))
  .add('with Icon', () => ({
    components: { GButton },
    template: `
      <div>
        <GButton size="big" icon="info"></GButton>
        <GButton size="medium" icon="info"></GButton>
        <GButton size="small" icon="info"></GButton>
      </div>
    `,
  }))
  .add('disabled', () => ({
    components: { GButton },
    template: `
      <GButton disabled>Disabled Button</GButton>
    `,
  }))
  .add('fluid', () => ({
    components: { GButton },
    template: `
      <GButton variant="primary" fluid leftIcon="info">Deneme</GButton>
    `,
  }))
  .add('gradient', () => ({
    components: { GButton },
    template: `
      <GButton variant="gradient-purple" leftIcon="play-circle">TUTORIAL TYPE</GButton>
    `,
  })).add('link', () => ({
    components: { GButton },
    template: `
    <GButton variant="link">Link</GButton>
  `,
  }));
