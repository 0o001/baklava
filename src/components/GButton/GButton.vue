<template>
  <button
    v-bind="$attrs"
    v-on="$listeners"
    :class="classNames"
  >
    <div :class="contentClassname">
      <GIcon
        :class="slotClassName"
        :size="iconSize"
        class="-leftIcon"
        v-if="leftIcon"
        :name="leftIcon"
        :color="iconColor"
      />
      <slot />
      <GIcon
        v-if="icon"
        :size="iconSize"
        :name="icon"
        :color="iconColor"
      />
      <GIcon
        :class="slotClassName"
        :size="iconSize"
        class="-rightIcon"
        v-if="rightIcon"
        :name="rightIcon"
        :color="iconColor"
      />
    </div>
  </button>
</template>

<script lang="ts">

import GIcon from '../GIcon/GIcon.vue';
import $_ from '../../utils/dashUtil.ts';

const VALID_BUTTON_VARIANT = [
  'passive',
  'primary',
  'primary-darker',
  'secondary',
  'secondary-darker',
  'tertiary',
  'tertiary-darker',
  'success',
  'success-darker',
  'info',
  'info-darker',
  'warning',
  'warning-darker',
  'danger',
  'danger-darker',
  'light',
  'light-darker',
  'dark',
  'dark-darker',
  'gradient-purple',
  'link',
];

export default {
  name: 'GButton',
  components: {
    GIcon,
  },
  props: {
    variant: {
      type: String,
      default: 'primary',
      validator: function (value: string) {
        if (!VALID_BUTTON_VARIANT.includes(value)) {
          throw new Error(`Invalid prop. Give of of ${VALID_BUTTON_VARIANT.join(',')}`);
        }

        return true;
      },
    },
    size: {
      type: String,
      default: 'big',
    },
    fluid: {
      type: Boolean,
      default: false,
    },
    leftIcon: {
      type: String,
      default: undefined,
    },
    rightIcon: {
      type: String,
      default: undefined,
    },
    icon: {
      type: String,
      default: undefined,
    },
    iconColor: {
      type: String,
      default: 'white',
    },
    outline: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classNames () {
      return {
        'g-button': true,
        '-fluid': this.fluid,
        '-icon': this.icon || this.leftIcon || this.rightIcon,
        '-onlyIcon': !!this.icon,
        [$_(this.variant)]: true,
        '-outline': this.outline,
      };
    },
    contentClassname () {
      return {
        content: true,
        [$_(this.size)]: true,
      };
    },
    slotClassName () {
      return {
        'g-mr-10': !!this.leftIcon || false,
        'g-ml-10': !!this.rightIcon || false,
      };
    },
    iconSize () {
      return {
        big: 20,
        medium: 16,
        small: 12,
      }[this.size];
    },
  },
};
</script>

<style lang="scss" scoped>
.g-button {
  display: inline-block;
  outline: none;
  border: 1px solid transparent;
  border-radius: var(--radius-md);
  padding: 0;

  &:hover {
    cursor: pointer;
  }

  &:disabled {
    opacity: .65;
    pointer-events: none;
    user-select: none;
  }

  .content {
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: var(--font-family-base);
    font-size: var(--font-size-14);
    font-weight: 500;
    line-height: 1;
  }

  &.-fluid {
    width: 100%;
    justify-content: center;
  }

  .-big {
    padding: 12px 30px;
  }

  .-medium {
    padding: 8px 20px;
  }

  .-small {
    padding: 5px 15px;
    font-size: var(--font-size-12);
  }

  &.-icon {
    .-big {
      padding: 10px 30px;
    }
    .-medium {
      padding: 8px 20px;
    }
  }

  &.-onlyIcon {
    line-height: 0;
    .-big {
      padding: 10px;
    }
    .-medium {
      padding: 8px;
    }
    .-small {
      padding: 6px;
    }
  }

  &.-passive {
    background-color: white;
    border: 1px solid var(--mid-grey-100);
    color: var(--light-grey-100);
    .-big {
      padding: 11px 29px;
    }
    .-middle {
      padding: 7px 19px;
    }
    .-small {
      padding: 4px 14px;
    }
  }

  &.-primary {
    background-color: var(--orange-500);
    color: var(--white);

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--orange-500);
      color: var(--orange-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--orange-500);
      }
    }

    &:hover {
      background-color: var(--orange-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-primary-darker {
    background-color: var(--orange-300);
    color: white;
    &:hover {
      background-color: var(--orange-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-secondary {
    background-color: var(--main-grey-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--main-grey-500);
      color: var(--main-grey-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--main-grey-500);
      }
    }

    &:hover {
      background-color: var(--main-grey-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-secondary-darker {
    background-color: var(--main-grey-300);
    color: white;
    &:hover {
      background-color: var(--main-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-tertiary {
    background-color: var(--light-grey-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--light-grey-500);
      color: var(--light-grey-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--light-grey-500);
      }
    }

    &:hover {
      background-color: var(--light-grey-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-tertiary-darker {
    background-color: var(--light-grey-300);
    color: white;
    &:hover {
      background-color: var(--light-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-success {
    background-color: var(--green-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--green-500);
      color: var(--green-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--green-500);
      }
    }

    &:hover {
      background-color: var(--green-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-success-darker {
    background-color: var(--green-300);
    color: white;
    &:hover {
      background-color: var(--green-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-info {
    background-color: var(--blue-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--blue-500);
      color: var(--blue-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--blue-500);
      }
    }

    &:hover {
      background-color: var(--blue-400);
    }
    &.icon {
      color: black;
    }
  }

  &.-info-darker {
    background-color: var(--blue-300);
    color: white;
    &:hover {
      background-color: var(--blue-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-warning {
    background-color: var(--orange-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--orange-500);
      color: var(--orange-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--orange-500);
      }
    }

    &:hover {
      background-color: var(--orange-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-warning-darker {
    background-color: var(--orange-300);
    color: white;
    &:hover {
      background-color: var(--orange-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-danger {
    background-color: var(--red-500);
    color: white;

    &.-outline {
      background-color: transparent;
      background-image: none;
      border-color: var(--red-500);
      color: var(--red-500);

      &:hover {
        color: var(--white);
      }
      &.icon {
        color: var(--red-500);
      }
    }

    &:hover {
      background-color: var(--red-400);
    }
    &.icon {
      color: white;
    }
  }

  &.-danger-darker {
    background-color: var(--red-300);
    color: white;
    &:hover {
      background-color: var(--red-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-light {
    background-color: var(--bg-grey-500);
    color: black;
    &:hover {
      background-color: var(--bg-grey-400);
    }
    &.icon {
      color: black;
      path {
        fill: black;
      }
    }
  }

  &.-light-darker {
    background-color: var(--bg-grey-300);
    color: white;
    &:hover {
      background-color: var(--bg-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-dark {
    background-color: var(--main-grey-500);
    color: white;
    &:hover {
      background-color: var(--main-grey-400);
    }
    &.icon {
      color: white;
      path {
        fill: white;
      }
    }
  }

  &.-link {
    background-color: transparent;
    color: var( --main-grey-500);
    text-decoration: underline;

    .content {
      font-weight: 400;
    }

    &:hover {
      text-decoration: underline;
    }
    &.icon {
      color: white;
      path {
        fill: white;
      }
    }
  }

  &.-dark-darker {
    background-color: var(--main-grey-300);
    color: white;
    &:hover {
      background-color: var(--main-grey-200);
    }
    &.icon {
      color: white;
    }
  }

  &.-gradient-purple {
    cursor: pointer;
    line-height: 1;
    display: inline-block;
    border: 1px solid transparent !important;
    padding: 0;
    border-radius: var(--radius-md);
    background-image: linear-gradient(180deg,var(--bg-grey-500),var(--bg-grey-500)),linear-gradient(100deg,var(--purple-500),var(--blue-500));
    background-origin: border-box;
    background-clip: content-box,border-box;
    text-decoration: none !important;
    svg {
      stroke: var(--purple-500);
    }
    .content {
      background-image: linear-gradient(96deg,var(--purple-500),var(--blue-500));
      -webkit-background-clip: text;
      background-clip: text;
      -webkit-text-fill-color: transparent;
    }
    &:hover {
      background-image: linear-gradient(280deg,var(--blue-500),var(--purple-500));
      .content {
        background-image: none;
        color: white;
        -webkit-text-fill-color: white;
      }
      svg {
        stroke: white;
      }
    }
    &.icon {
      color: white;
      path {
        fill: white;
      }
    }
  }
}
</style>
