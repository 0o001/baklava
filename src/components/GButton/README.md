# GButton

## Slots

| Name      | Description |
| --------- | ----------- |
| `default` | &nbsp;      |

## Props

| Name         | Type      | Description | Default     |
| ------------ | --------- | ----------- | ----------- |
| `variant`    | `String`  |             | `"primary"` |
| `size`       | `String`  |             | `"big"`     |
| `fluid`      | `Boolean` |             | `false`     |
| `left-icon`  | `String`  |             | `undefined` |
| `right-icon` | `String`  |             | `undefined` |
| `icon`       | `String`  |             | `undefined` |
| `icon-color` | `String`  |             | `"white"`   |
| `outline`    | `Boolean` |             | `false`     |

## Computed Properties

| Name               | Type     | Description                                                                      |
| ------------------ | -------- | -------------------------------------------------------------------------------- |
| `classNames`       | `object` | **Dependencies:** `fluid`, `icon`, `leftIcon`, `rightIcon`, `variant`, `outline` |
| `contentClassname` | `object` | **Dependencies:** `size`                                                         |
| `slotClassName`    | `object` | **Dependencies:** `leftIcon`, `rightIcon`                                        |
| `iconSize`         | `object` | **Dependencies:** `size`                                                         |

