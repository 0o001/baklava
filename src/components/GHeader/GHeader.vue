<template>
  <GBox
    flex
    justify="space-between"
    :my="32"
  >
    <GBox
      flex
      direction="row"
    >
      <slot name="title-before" />
      <GBox
        flex
        direction="column"
        :mr="30"
      >
        <GText variant="headline-03">
          {{ title }}
        </GText>
        <slot name="breadcrumb" />
      </GBox>
      <slot name="left-area" />
    </GBox>
    <GBox box>
      <slot name="right-area" />
    </GBox>
  </GBox>
</template>

<script lang="ts">

import GBox from '../GBox/GBox.vue';
import GText from '../GText/GText.vue';

export default {
  name: 'GHeader',
  components: {
    GBox,
    GText,
  },
  props: {
    title: {
      type: String,
      default: null,
    },
  },
};
</script>
