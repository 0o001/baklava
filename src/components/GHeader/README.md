# GHeader

## Slots

| Name           | Description |
| -------------- | ----------- |
| `title-before` |             |
| `breadcrumb`   |             |
| `left-area`    |             |
| `right-area`   | &nbsp;      |

## Props

| Name    | Type     | Description | Default |
| ------- | -------- | ----------- | ------- |
| `title` | `String` |             | `null`  |

