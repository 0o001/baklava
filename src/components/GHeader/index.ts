import GHeader from './GHeader.vue';

export default GHeader;
