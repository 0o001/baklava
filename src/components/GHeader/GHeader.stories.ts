import { storiesOf } from '@storybook/vue';
import GHeader from './';
import GButton from '../GButton';
import GText from '../GText';
import GIcon from '../GIcon';
import GBreadcrumb from '../GBreadcrumb';
import { action } from '@storybook/addon-actions';

storiesOf('GHeader', module)
  .addParameters({ component: GHeader })
  .add('base', () => ({
    components: { GHeader },
    template: '<GHeader title="Kampanyalarım"></GHeader>',
  }))
  .add('with right button', () => ({
    components: { GHeader, GButton },
    template: `
      <GHeader title="Kampanyalarım">
        <template #right-area>
          <GButton variant="gradient-purple" leftIcon="play-circle">
            RIGHT BUTTON
          </GButton>
        </template>
      </GHeader>
    `,
  }))
  .add('with left button', () => ({
    components: { GHeader, GButton },
    template: `
      <GHeader title="Kampanyalarım">
        <template #left-area>
          <GButton variant="gradient-purple" leftIcon="play-circle">
            LEFT BUTTON
          </GButton>
        </template>
      </GHeader>
    `,
  }))
  .add('with title before slot', () => ({
    components: { GHeader, GIcon },
    template: `
      <GHeader title="Kampanyalarım">
        <template #title-before>
          <GIcon name="arrow-left" size="32px"/>
        </template>
      </GHeader>
    `,
  }))
  .add('with both side buttons', () => ({
    components: { GHeader, GButton },
    template: `
      <GHeader title="Kampanyalarım">
        <template #left-area>
          <GButton variant="gradient-purple" leftIcon="play-circle">
            LEFT BUTTON
          </GButton>
        </template>
        <template #right-area>
          <GButton variant="gradient-purple" leftIcon="play-circle">
            RIGHT BUTTON
          </GButton>
        </template>
      </GHeader>
    `,
  }))
  .add('with breadcrumb and no buttons', () => ({
    components: { GHeader, GButton, GText, GBreadcrumb },
    template: `
      <GHeader title="Kampanyalarım">
        <template #breadcrumb>
          <GBreadcrumb :items="items" @click="onClick"></GBreadcrumb>
        </template>
      </GHeader>
    `,
    computed: {
      items () {
        return [
          { text: 'Anasayfa', href: '/' },
          { text: 'Page', href: '/a' },
        ];
      },
    },
    methods: {
      onClick (item) {
        action('Click a link from breadcrumb')(item);
      },
    },
  }))
  .add('with breadcrumb and right button', () => ({
    components: { GHeader, GButton, GText, GBreadcrumb },
    template: `
      <GHeader title="Kampanyalarım">
        <template #breadcrumb>
          <GBreadcrumb :items="items" @click="onClick"></GBreadcrumb>
        </template>
        <template #right-area>
          <GButton variant="gradient-purple" leftIcon="play-circle">
            RIGHT BUTTON
          </GButton>
        </template>
      </GHeader>
    `,
    computed: {
      items () {
        return [
          { text: 'Anasayfa', href: '/' },
          { text: 'Page', href: '/a' },
        ];
      },
    },
    methods: {
      onClick (item) {
        action('Click a link from breadcrumb')(item);
      },
    },
  }));
