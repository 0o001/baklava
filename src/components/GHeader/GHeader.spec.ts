import { mount } from '@vue/test-utils';
import GHeader from './';
import GButton from '../GButton';

describe('GHeader', () => {
  let wrapper: any;

  it('should match snapshot when show is prop is given as false ', () => {
    wrapper = mount(GHeader);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render header with given title and right-area', async () => {
    // when
    wrapper = mount(GHeader, {
      propsData: {
        title: 'Kampanyalarım',
      },
      slots: {
        'right-area': `<GButton variant="gradient-purple" leftIcon="play-circle">RIGHT BUTTON</GButton>`
      },
      stubs: {
        'GButton': GButton
      },
    });

    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.title).toEqual('Kampanyalarım');
    expect(wrapper.vm.$slots['right-area'][0].componentOptions.children[0].text).toEqual('RIGHT BUTTON');
    expect(wrapper.vm.$slots['left-area']).toBeUndefined();
    expect(wrapper.vm.$slots['breadcrumb']).toBeUndefined();
  });

  it('should render header with given title, right-area and left-area', async () => {
    // when
    wrapper = mount(GHeader, {
      propsData: {
        title: 'Kampanyalarım',
      },
      slots: {
        'right-area': `<GButton variant="gradient-purple" leftIcon="play-circle">RIGHT BUTTON</GButton>`,
        'left-area': `<GButton variant="gradient-purple" leftIcon="play-circle">LEFT BUTTON</GButton>`
      },
      stubs: {
        'GButton': GButton
      },
    });

    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.title).toEqual('Kampanyalarım');
    expect(wrapper.vm.$slots['right-area'][0].componentOptions.children[0].text).toEqual('RIGHT BUTTON')
    expect(wrapper.vm.$slots['left-area'][0].componentOptions.children[0].text).toEqual('LEFT BUTTON')
    expect(wrapper.vm.$slots['breadcrumb']).toBeUndefined();
  });

  it('should render header with given title, title-before', async () => {
    // when
    wrapper = mount(GHeader, {
      propsData: {
        title: 'Kampanyalarım',
      },
      slots: {
        'title-before': `<GButton variant="gradient-purple" leftIcon="play-circle">TITLE BEFORE</GButton>`
      },
      stubs: {
        'GButton': GButton
      },
    });

    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.title).toEqual('Kampanyalarım');
    expect(wrapper.vm.$slots['title-before'][0].componentOptions.children[0].text).toEqual('TITLE BEFORE')
    expect(wrapper.vm.$slots['breadcrumb']).toBeUndefined();
  });
});
