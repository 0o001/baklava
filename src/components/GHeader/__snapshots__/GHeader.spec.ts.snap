// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GHeader should match snapshot when show is prop is given as false  1`] = `
<div
  class="g-d-flex g-d-justify-space-between g-my-32"
>
  <div
    class="g-d-flex g-d-direction-row"
  >
     
    <div
      class="g-d-flex g-d-direction-column g-mr-30"
    >
      <div
        class="g-text-main-grey-500 -headline-03 g-text"
      >
        
        
      
      </div>
       
    </div>
     
  </div>
   
  <div
    box=""
    class=""
  />
</div>
`;
